woocommerce/woocommerce-admin
# Onboarding: Add tax task.

Fixes #2683

Adds a task configuration task to the task list.

Note that this will require a follow-up to check for activated plugins and jetpack connection via the wc-api.

### Screenshots
<img width="708" alt="Screen Shot 2019-08-22 at 5 37 53 PM" src="https://user-images.githubusercontent.com/10561050/63504715-b349a580-c504-11e9-886b-15d81a69bcf9.png">
<img width="725" alt="Screen Shot 2019-08-22 at 5 37 42 PM" src="https://user-images.githubusercontent.com/10561050/63504717-b349a580-c504-11e9-8681-1ddaf916173c.png">


### Detailed test instructions:

1. Delete your store address (or at least the `woocommerce_store_address` option), deactivate/uninstall wc-services and jetpack.
2. Go to the Tax task in the task list.
3.  Go through the steps and make sure you can fill in your address and install the 2 required plugins.
4.  The Jetpack wc-api connection will be added in a follow-up, for now toggle the value of `isJetpackConnected` in `tax.js` to bypass this step.
5.  Set `isJetpackConnected` to true and `pluginsToActivate` to an empty array.  Refresh this task and if you have all requirements met, you should see the success screen instead of the stepper.
6.  Make sure add a non-supported tax country forces you to manual configuration. https://developers.taxjar.com/api/reference/#countries
7. Make sure the toggle in the 4th step of the Stepper or the buttons in the success screen change the value of `wc_connect_taxes_enabled` in `wp_options`.  (Note: Jetpack and WCS must be installed and connected for this option to work via the WC REST API).
